### PR TITLE
ci: use docker/build-push-action for image pushes

### PR DESCRIPTION
**Pull request by jackfrancis**
This PR delivers a "release-image" github action workflow to automatically publish

### CODE REVIEW

File: .github/workflows/release-image.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.CR_PAT }}
+      - name: build and push
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          file: 'vmss-prototype/Dockerfile'
+          tags: |
+            ghcr.io/owner/templated-repo:${{ env.RELEASE_VERSION }}
```
Comment (Michael-Sinz) [inline .github/workflows/release-image.yml]: Does this replace owner/templated-repo somehow or is that just placeholder for us to put the right thing?

Comment (Michael-Sinz) [reply .github/workflows/release-image.yml]: Looking on the web:  https://github.com/docker/build-push-action

Comment (jackfrancis) [reply .github/workflows/release-image.yml]: You found the bug, good job :)

File: .github/workflows/release-image.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.CR_PAT }}
+      - name: build and push
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          file: 'vmss-prototype/Dockerfile'
```
Comment (Michael-Sinz) [inline .github/workflows/release-image.yml]: I think we need a context of 'vmss-prototype'

Unsure but the way I read the docs we need to pick that (albeit it is strange you can't just give a context and default the docker file)

Comment (Michael-Sinz) [reply .github/workflows/release-image.yml]: Also, we may wish to change the name (in the future) of this directory given that the directory and the binary have the same name and could cause confusion for certain types of errors.

Comment (Michael-Sinz) [reply .github/workflows/release-image.yml]: https://github.com/docker/build-push-action#inputs

Comment (jackfrancis) [reply .github/workflows/release-image.yml]: added a path context

**Comment by Michael-Sinz** [comment]
Looks like a possible good starting point - I am not super familiar with github so I can't tell for sure.

/lgtm

**Pull request merged**